From cb6dc7c82fea4b624bdb4f875ed051413b62ffb5 Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Tue, 21 Mar 2017 13:28:38 -0400
Subject: [PATCH 58/89] arm: dts: k2: set dma-coherent for sa subsystem

We moved crypto node one level down in dts. To keep its dma properties,
we need to set dma-coherent and dma-ranges properties for its parent.

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 arch/arm/boot/dts/keystone-k2e-netcp.dtsi  | 2 ++
 arch/arm/boot/dts/keystone-k2hk-netcp.dtsi | 2 ++
 arch/arm/boot/dts/keystone-k2l-netcp.dtsi  | 2 ++
 3 files changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/keystone-k2e-netcp.dtsi b/arch/arm/boot/dts/keystone-k2e-netcp.dtsi
index a388a8c..01acfa7 100644
--- a/arch/arm/boot/dts/keystone-k2e-netcp.dtsi
+++ b/arch/arm/boot/dts/keystone-k2e-netcp.dtsi
@@ -763,6 +763,8 @@ sa_subsys: subsys@24080000 {
 	#size-cells = <1>;
 	ranges = <0 0x24080000 0x40000>;
 	compatible = "simple-bus";
+	dma-coherent;
+	dma-ranges;
 
 	sa_config: subsys@0 {
 		compatible      = "syscon";
diff --git a/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi b/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
index dee2e17..f51d20b 100644
--- a/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
+++ b/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
@@ -702,6 +702,8 @@ sa_subsys: subsys@20c0000 {
 	#size-cells = <1>;
 	ranges = <0 0x20c0000 0x40000>;
 	compatible = "simple-bus";
+	dma-coherent;
+	dma-ranges;
 
 	sa_config: subsys@0 {
 		compatible      = "syscon";
diff --git a/arch/arm/boot/dts/keystone-k2l-netcp.dtsi b/arch/arm/boot/dts/keystone-k2l-netcp.dtsi
index 6216bea..0991bef 100644
--- a/arch/arm/boot/dts/keystone-k2l-netcp.dtsi
+++ b/arch/arm/boot/dts/keystone-k2l-netcp.dtsi
@@ -541,6 +541,8 @@ sa_subsys: subsys@26080000 {
 	#size-cells = <1>;
 	ranges = <0 0x26080000 0x40000>;
 	compatible = "simple-bus";
+	dma-coherent;
+	dma-ranges;
 
 	sa_config: subsys@0 {
 		compatible      = "syscon";
-- 
1.9.1

